cmake/platform/types/PlatformTypes.hpp

```diff
@@ -38,10 +38,11 @@
 #include <cinttypes>
 #include <limits>
 
-#define FW_HAS_64_BIT    1 //!< Architecture supports 64 bit integers
-#define FW_HAS_32_BIT    1 //!< Architecture supports 32 bit integers
-#define FW_HAS_16_BIT    1 //!< Architecture supports 16 bit integers
-#define FW_HAS_F64    1 //!< Architecture supports 64 bit floating point numbers
+#define FW_HAS_64_BIT 1  //!< Architecture supports 64 bit integers
+#define FW_HAS_32_BIT 1  //!< Architecture supports 32 bit integers
+#define FW_HAS_16_BIT 1  //!< Architecture supports 16 bit integers
+#define FW_HAS_F64 1     //!< Architecture supports 64 bit floating point numbers
+#define SKIP_FLOAT_IEEE_754_COMPLIANCE 0  //!<  Check IEEE 754 compliance of floating point arithmetic
```

Comment:
This will skip the check, as `#ifndef` checks for definitions not value of the definition.  We have two options:

1. (Easy) remove this definition.  The check will run.
2. (More clear) Changes the check in `BasicTypes.hpp` to: `#if !defined(SKIP_FLOAT_IEEE_754_COMPLIANCE) || !SKIP_FLOAT_IEEE_754_COMPLIANCE`

Reply:
The behavior is of the easy option is:

1. Value undefined: run check
2. Value defined: skip check

The behavior of the clear option is:
1. Value undefined: run check
2. Value defined: run check when not true

In either case, we want the undefined behavior to check and only allow the check to skip when specifically turned off.  The more clear option allows users to define the field but still run the check with a false.

Reply:
"Clear option" done!

Reply:
Looks good now.  Configuration allows setting, unsetting, and defaulting to the check!